### DISCUSSION TITLE
Replying to thread while review is pending is confusing

### DISCUSSION THREAD

**Discussion by arlowhite** [community/community#16396] [Pull Requests]
1. In my PR I started a review and left it pending.
2. Later, in the Conversation tab I replied to a thread, this reply was marked Pending
3. Realizing my reply was associated with the pending review I submitted the review so the other person could see my reply.

This is where it gets confusing:
Now I have two copies of that reply in the Conversation tab. One is associated with the review section and the other is on the thread I replied to above. I'm not certain if these are separate or refer to the same thing so I test it: I edit the one in the review section; it changes text separately so I assume they're separate entities. Now I delete the comment in the review section, but it deletes both! I've lost my thread reply and can't get it back.

I don't think thread replies should be associated with your pending review. They should just immediately append to the thread regardless of if you have a review in progress.

**Comment by winterqt** [comment]
This is also very confusing when the comments show up associated with the review, but there's no reply UI like new comments, in addition to it showing up in the thread.

Something I would like to know if is this was a deliberate design decision, or just a bug. If the former, this should be made clearer, and if the latter, it should (obviously) be fixed.

**Reply by marco-ybz** [reply]
oo

**Comment by tonglil** [comment]
Agreed, replying to a thread in the middle of a review should be immediate, and not as a part of the "review".

Just because I happen upon a thread during my review, doesn't mean my response to that thread is related to my review.
I want to respond to the thread _at that moment in time_, not have to come back to it after reviewing the rest of the PR, at which point I might have forgotten about what I wanted to say.

**Comment by ian-h-chamberlain** [comment]
Seriously, this is one of my biggest frustrations with the GitHub PR review process — I'm surprised I can't find more issues related to this! Showing two copies of each comment is confusing and wastes time during a review.

I think there are a couple of options that would make this feel more intuitive / not-broken:

* When replying to a thread, maybe provide the option to "add to review" or "comment immediately"? I agree that defaulting to reply immediately would make more sense but perhaps there are some cases where people do want to wait until they submit their review
* Don't show "review replies" under the review at all. There is no reply button, the comment is already shown once (inline in the thread), and this extra display serves no purpose (and is often confusing! If the comment is referencing its parent, there is no context to understand what it's talking about)

A related issue is #10369 which has a lot more votes, but discusses something related — the fact that comments while a review is pending do not show up immediately. Perhaps if that were solved it would be a "two birds, one stone" kind of thing, since the replies would no longer be considered "part of the review" maybe they wouldn't have that second copy anymore.

Unfortunately it looks like that issue has been stagnant for quite a while, so not too optimistic this will be improved anytime soon 😢

**Comment by sophiasmithastri** [comment]
if you need a reply to show up immediately, submit your pending review first, or avoid starting a review until you finish the discussion thread. I also like the idea of an explicit choice like “comment now” vs “include in review,” because the current behavior feels like a UI bug, not a feature.

**Comment by Jacob-Stevens-Haas** [comment]
Wish I could dismiss a pending reivew so I could just comment normally.
